internal/client/repository.go

```diff
@@ -311,8 +311,9 @@ func (t *tags) Get(ctx context.Context, tag string) (v1.Descriptor, error) {
 			return nil, err
 		}
 
-		for _, t := range distribution.ManifestMediaTypes() {
-			req.Header.Add("Accept", t)
+		mediaTypes := distribution.ManifestMediaTypes()
+		if len(mediaTypes) > 0 {
+			req.Header.Add("Accept", strings.Join(mediaTypes, ", "))
```

Comment:
It feels like we're working around go stdlib behavior here (which allows multiple values); is this something that should be fixed there instead?

Reply:
In most cases, there's no issue. However, when the client goes through an intermediary proxy, if the `Accept` header is an array, some proxies will only process the first header. Making it comma-separated would be more universal.